src/main/cli/cli.c

```diff
@@ -6772,42 +6784,78 @@ static void processCharacterInteractive(const char c)
     }
 }
 
-void cliProcess(void)
+bool cliProcess(void)
 {
     if (!cliWriter || !cliMode) {
-        return;
+        return false;
     }
 
-    // Flush the buffer to get rid of any MSP data polls sent by configurator after CLI was invoked
-    cliWriterFlush();
-
     while (serialRxBytesWaiting(cliPort)) {
         uint8_t c = serialRead(cliPort);
+        if (cliInteractive) {
+            processCharacterInteractive(c);
+        } else {
+            // handle terminating flow control character
+            if (c == 0x3 || (cmp32(millis(), cliEntryTime) > 2000)) { // CTRL-C (ETX) or 2 seconds timeout
+                cliWrite(0x3); // send end of text, terminating flow control
+                cliExit(false);
+                return cliMode;
+            }
+            processCharacter(c);
+        }
+    }
+    cliWriterFlush();
+    return cliMode;
+}
+
+static void cliExit(const bool reboot)
+{
+    cliWriterFlush();
+    waitForSerialPortToFinishTransmitting(cliPort);
+    memset(cliBuffer, 0, sizeof(cliBuffer));
```

Comment:
(I'd move(or copy) this buffer zeroing to `cliEnter`, to start each command in identical state. Or refactor it cliClearInput, is is used in 3 places now)